dvc/repo/add.py

```diff
@@ -58,7 +59,8 @@ def _create_stages(repo, targets, fname, no_commit):
 
     with repo.state:
         for out in targets:
-            stage = Stage.create(repo=repo, outs=[out], add=True, fname=fname)
+            outs = loads_from([out])
```

Comment:
same here, I think all this stuff loads, etc should be hidded in the Stage itself. If needs to understand multiple formats let's make some internal helpers. Otherwise we have these loads for outputs and dependencies in multiple places.